src/UniversalRewardsDistributor.sol

```diff
@@ -143,30 +130,38 @@ contract UniversalRewardsDistributor is IUniversalRewardsDistributor {
     /// @notice Creates a new distribution.
     /// @param initialTimelock The initial timelock for the new distribution.
     /// @param initialRoot The initial merkle tree's root for the new distribution.
-    /// @dev The caller of this function is the owner and the treasury of the new distribution.
-    function createDistribution(uint256 initialTimelock, bytes32 initialRoot)
-        external
-        returns (uint256 distributionId)
-    {
+    /// @param initialOwner The initial owner for the new distribution.
+    /// @param initialPendingTreasury The initial pending treasury for the new distribution.
+    /// @dev the initial treasury is always msg.sender. The initialPendingTreasury can be set to address(0).
+    function createDistribution(
+        uint256 initialTimelock,
+        bytes32 initialRoot,
+        address initialOwner,
+        address initialPendingTreasury
+    ) external returns (uint256 distributionId) {
+        address owner = initialOwner == address(0) ? msg.sender : initialOwner;
+
         distributionId = nextDistributionId++;
-        ownerOf[distributionId] = msg.sender;
+        ownerOf[distributionId] = owner;
         treasuryOf[distributionId] = msg.sender;
```

Comment:
Maybe we should log `TreasuryUpdated`

Reply:
IMO no, the treasury is setted up, not updated

Reply:
ok but shouldn't we log such storage change still? Same question for the owner. This is what OZ is doing and what we did in the past.